src/frontend/screens/AuthScreen.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from "react";
+import { defineMessages, FormattedMessage } from "react-intl";
+import { View, Image, Text, StyleSheet } from "react-native";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import { DARK_BLUE, DARK_GREY, MEDIUM_GREY, WARNING_RED } from "../lib/styles";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
+import { KILL_PASSCODE, PASSWORD_KEY } from "../constants";
+import { AuthState, SecurityContext } from "../context/SecurityContext";
+import { PasswordInput } from "../sharedComponents/PasswordInput";
+import { useNavigation } from "react-navigation-hooks";
+
+const m = defineMessages({
+  enterPass: {
+    id: "screens.EnterPassword.enterPass",
+    defaultMessage: "Enter your password",
+  },
+  wrongPass: {
+    id: "screens.EnterPassword.wrongPass",
+    defaultMessage: "Incorrect password, please try again ",
+  },
+});
+
+export const AuthScreen: NavigationStackScreenComponent = () => {
+  const [error, setError] = React.useState(false);
+  const [authState, setAuthState] = React.useContext(SecurityContext);
+
+  const { navigate } = useNavigation();
+
+  async function validatePass(inputtedPass: string, clearInput: () => void) {
+    console.log("input" + inputtedPass);
+    console.log("currentPass" + authState.passcode);
+    console.log(await AsyncStorage.getItem(PASSWORD_KEY));
+    if (inputtedPass === KILL_PASSCODE && authState.killModeEnabled) {
+      setAuthState({ type: "appMode:set", newAppMode: "kill" });
+      setAuthState({
+        type: "setAuthStatus",
+        newAuthStatus: "authenticated",
+      });
+      return;
+    }
```

Comment:
not a blocker, but would it make sense to always go to the map screen after entering the kill passcode instead of the last visited one? maybe it doesn't but just something i wanted to pose.

not sure where that logic would live but reading this got me thinking about that